app-modules/inventory-management/src/Filament/Actions/CheckInAssetHeaderAction.php

```diff
@@ -87,16 +89,30 @@ protected function setUp(): void
                 $this->failure();
             }
 
-            $asset->checkIns()->create([
+            //Before this change, $createArray was not a varaible but just defined in the $asset->checkOuts()->create() call on line 115
+            //When this feature flag is cleaned up, it can return to being like that (i.e., replace $createArray with lines 95-103)
+            $createArray = MakeContactNotPolymorphicFeature::active() ?
+            [
                 'checked_in_by_type' => auth()->user()?->getMorphClass(),
                 'checked_in_by_id' => auth()->user()?->id,
                 // TODO Should this always simply be the latest check out, or do we want to support
                 // The possibility that the person checking in is different than whoever checked out?
-                'checked_in_from_type' => $asset->latestCheckOut->checked_out_to_type,
                 'checked_in_from_id' => $asset->latestCheckOut->checked_out_to_id,
                 'notes' => $data['notes'],
                 'checked_in_at' => $data['checked_in_at'] ?? now(),
-            ]);
+            ] :
+            [
+                'checked_in_by_type' => auth()->user()?->getMorphClass(),
+                'checked_in_by_id' => auth()->user()?->id,
+                // TODO Should this always simply be the latest check out, or do we want to support
+                // The possibility that the person checking in is different than whoever checked out?
```

Comment:
Though I think the logic here is sound. I believe for now we always want it to be the person that checked it out. So this comment can be removed.